shared/accuracy_test.h

```diff
@@ -859,6 +859,20 @@ inline void fft_vs_reference_impl(Tparams& params, bool round_trip)
             cpu_input.swap(last_cpu_fft_data.cpu_input);
             cpu_output.swap(last_cpu_fft_data.cpu_output);
             run_fftw = false;
+            if(params.scale_factor != 1.0)
```

Comment:
Another approach would be to add `scale_factor` to `last_cpu_fft_data`, similar to how it already tracks `run_callbacks`.  This would mean the cpu output is still stored scaled in the cache, but at least the logic that manages the cache is aware that this has happened.

IMO that would be simpler than trying to undo the result scaling?